### PR TITLE
03-Bash homework

### PR DESCRIPTION
**Pull request by serega911**
03-Bash homework

Could you please take a look at my homework

Signed-off-by: Serega911 <pasechnuy.sergey@gmail.com>

### CODE REVIEW

File: 03-bash/task1.sh [hunk 1]
```diff
@@ -0,0 +1,190 @@
+#!/bin/bash
+
+is_integer () {
+	local re='^[0-9]+$'
+	[[ $1 =~ $re ]]
+	return $?
```
Comment (yekovalyov) [inline 03-bash/task1.sh]: return $? is not required. $? is returned by default

File: 03-bash/task1.sh [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+		echo 'Usage:'
+		echo '  ./task1.sh [-h|--help] [-n num] [file]'
+		echo '     num --  amount of files'
+		echo '     file -- path to file to store results'
+	fi
+	exit $1
+}
+
+err_not_a_number() {
+	if is_ukr_locale; then
```
Comment (yekovalyov) [inline 03-bash/task1.sh]: multiple calls of is_ukr_locale.
It is better to save value as variable instead calling function.
Also it is better save localized test as variable.

File: 03-bash/task1.sh [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+integer_to_zero_string () {
+	local ret
+	printf -v ret "%04d" $1
+	echo "$ret"
+}
+
+is_ukr_locale() {
+	local re=$(locale | grep 'LANG=uk_UA')
+	if [ "$re" == '' ]; then
```
Comment (yekovalyov) [inline 03-bash/task1.sh]: -z is usual way to compare with empty line

File: 03-bash/task1.sh [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+		return 1
```
Comment (yekovalyov) [inline 03-bash/task1.sh]: It is possible to change 5 line if...else ...fi to only line with compare

File: 03-bash/task1.sh (hunk 1 above)
Comment (serega911) [reply 03-bash/task1.sh]: As I understand $? is a result of [[ $1 =~ $re ]] command. So, I prefer to return explicit value

File: 03-bash/task1.sh (hunk 4 above)
Comment (serega911) [reply 03-bash/task1.sh]: Could you explain how can I do this, please.

File: 03-bash/task1.sh (hunk 3 above)
Comment (serega911) [reply 03-bash/task1.sh]: done

**Pull request merged**